### PR TITLE
UIManager: get rid of self._running and self._run_forever

### PR DESCRIPTION
**Pull request by zwim**
@NiLuJe Is this what you wished in #9657?

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/9669)

### CODE REVIEW

File: frontend/ui/uimanager.lua [hunk 1]
```diff
@@ -68,7 +67,7 @@ function UIManager:init()
         self:nextTick(function()
             Device:saveSettings()
             Device:powerOff()
-            self:quit(Device:isKobo() and 88)
+            self:quit(Device:isKobo() and 88 or nil)
```
Comment (NiLuJe) [inline frontend/ui/uimanager.lua:70]: Oh, right, good point, devcaps eval to true or false, not true or nil ;).

**Review by NiLuJe — APPROVED** [review]
Yup, :lgtm:, thanks!

File: frontend/ui/uimanager.lua [hunk 2]
```diff
@@ -749,6 +748,18 @@ function UIManager:quit(exit_code)
...
+
+-- A simple wrapper for UIManager:quit()
+-- This may be overwritten by setRunForeverMode(); for testing purposes
+function UIManager:_gated_quit(exit_code)
+    return self:quit(exit_code)
+end
+
+-- Disable automatic UIManager quit; for testing purposes
+function UIManager:setRunForeverMode()
+    self._gated_quit = function() return false end
```
Comment (NiLuJe) [inline frontend/ui/uimanager.lua:762]: I don't recall if the testsuite ever unloads UIManager, so, we *may* need a way to restore the standard wrapper, too?

File: frontend/ui/uimanager.lua (hunk 1 above)
Comment (zwim) [reply frontend/ui/uimanager.lua:70]: In this special case I would prefer to write:
```
if Device:isKobo() then
    self:quit(88)
else
    self.quit()
end
```

File: frontend/ui/uimanager.lua (hunk 2 above)
Comment (zwim) [reply frontend/ui/uimanager.lua:762]: The test suite does a `package.unloadAll()` on most (all?) setup functions. On my laptop and CircleCi finish without error. 
So the answer is probably: No, we don't _need_ to have a restore functionallity.
(But I will implement these three lines for completeness.)

**Comment by zwim** [comment]
Ready to merge?

**Review by NiLuJe — CHANGES_REQUESTED** [review]
Goddamit GH review UI. Forgot to hit the big green button yet again. -_-".

**Review by NiLuJe — APPROVED** [review]

**Pull request merged**